File: dropshifter/dropshifter/report/supplier_order_confirmation_report/supplier_order_confirmation_report.py
# Copyright (c) 2023, MD Faiyaz Ansari and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe import _

def execute(filters=None):
    if not filters: filters = {}

    columns = get_columns()
    data = get_data(filters)

    return columns, data

def get_columns():
    return [
        _("Purchase Order") + ":Link/Purchase Order:120",
        _("Delivery Due Date") + ":Date:120",
        _("Status") + ":Data:120",
      
        _("Comment") + ":Data:300",  # Adding comment here
    ]

def get_data(filters):
    conditions = get_conditions(filters)

    return frappe.db.sql("""
        SELECT
            purchase_order,
            delivery_due_date,
            status,
            comment  # Adding comment here
        FROM
            `tabSupplier Order Confirmation`
        WHERE
            docstatus < 2 %s
        ORDER BY
            modified desc
    """ % conditions, filters, as_dict=1)


def get_conditions(filters):
    conditions = ""
    if filters.get("purchase_order"): conditions += " and purchase_order=%(purchase_order)s"
    if filters.get("delivery_due_date"): conditions += " and delivery_due_date=%(delivery_due_date)s"
    if filters.get("status"): conditions += " and status=%(status)s"
    if filters.get("comment"): conditions += " and comment=%(comment)s"  # Adding condition for comment

    return conditions

File: dropshifter/dropshifter/report/supplier_order_confirmation_report/supplier_order_confirmation_report.js
// Copyright (c) 2023, MD Faiyaz Ansari and contributors
// For license information, please see license.txt
/* eslint-disable */

// frappe.query_reports["Supplier Order Confirmation Report"] = {
// 	"filters": [

// 	]
// };
frappe.query_reports["Supplier Order Confirmation Report"] = {
    "filters": [
        {
            "fieldname":"purchase_order",
            "label": __("Purchase Order"),
            "fieldtype": "Link",
            "options": "Purchase Order"
        },
        {
            "fieldname":"delivery_due_date",
            "label": __("Delivery Due Date"),
            "fieldtype": "Date"
        },
        {
            "fieldname":"status",
            "label": __("Status"),
            "fieldtype": "Select",
            "options": "\nApproved\nCancelled\nHold"
        },
        // {
        //     "fieldname":"comment",
        //     "label": __("Comment"),
        //     "fieldtype": "Data"
        // }
    ]
}
